ext/lib/mgmt/mcumgr/samples/smp_svr/zephyr/prj.conf

```diff
@@ -0,0 +1,66 @@
+# Enable mcumgr.
+CONFIG_MCUMGR=y
+
+# Some command handlers require a large stack.
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=2048
+
+# Ensure an MCUboot-compatible binary is generated.
+CONFIG_BOOTLOADER_MCUBOOT=y
+
+# Increase the number of transmit buffers to avoid the Bluetooth stack buffer
+# leak.
+CONFIG_BT_L2CAP_TX_BUF_COUNT=8
+
+# Allow for large Bluetooth data packets.
+CONFIG_BT_L2CAP_TX_MTU=260
+CONFIG_BT_RX_BUF_LEN=260
+
+# Enable the Bluetooth and shell mcumgr transports.
+CONFIG_MCUMGR_SMP_BT=y
+CONFIG_MCUMGR_SMP_SHELL=y
+
+# Enable flash operations.
+CONFIG_FLASH=y
+
+# Enable the NFFS file system.
+CONFIG_FILE_SYSTEM=y
+CONFIG_FILE_SYSTEM_NFFS=y
+
+# Enable the flash circular buffer (FCB) for the reboot log.
+CONFIG_FLASH_PAGE_LAYOUT=y
+CONFIG_FLASH_MAP=y
+CONFIG_FCB=y
+
+# Enable the managed log (mdlog).
+CONFIG_MDLOG=y
+CONFIG_MDLOG_CONSOLE=y
+CONFIG_MDLOG_FCB=y
+
+# Required by the `taskstat` command.
+CONFIG_THREAD_MONITOR=y
+
+# Enable the reboot log.
+CONFIG_REBOOT_LOG=y
+
+# Enable assertions.  Required for crash logging.
+CONFIG_ASSERT=y
+
+# Enable statistics and statistic names.
+CONFIG_STATS=y
+CONFIG_STATS_NAMES=y
+
+# Enable all core commands.
+CONFIG_MCUMGR_CMD_FS_MGMT=y
+CONFIG_MCUMGR_CMD_IMG_MGMT=y
+CONFIG_MCUMGR_CMD_LOG_MGMT=y
+CONFIG_MCUMGR_CMD_OS_MGMT=y
+CONFIG_MCUMGR_CMD_STAT_MGMT=y
+
+### nRF5 specific settings
+
+# Disable flash-radio sync.  This is necessary for writing to the reboot log
+# with an interrupt handler.
+CONFIG_SOC_FLASH_NRF5_RADIO_SYNC=n
```

Comment:
Were is the problem - this should be enable whenever you want to write to flash while BLE is operating.
cc @cvinayak

Reply:
It has been a little while since I tried this, so I may have some details wrong.  My recollection is that the flash write routine was hanging when it attempted to acquire a semaphore with a timeout.  It seemed plausible enough that a timeout will never expire within an ISR, though I don't think I investigated this as much as I could have.  Disabling the flash radio sync setting prevented the hang.

Reply:
For sure: CONFIG_SOC_FLASH_NRF5_RADIO_SYNC must selected while you want to write to flash along with BLE connection.

Reply:
Can you try to increase by 2 macro FLASH_TIMEOUT_MS in drivers/flash/soc_flash_nrf5.c file.